### PR TITLE
fix InMemory read method

### PR DESCRIPTION
**Pull request by ellisjoe**

### CODE REVIEW

File: seek-io/src/main/java/com/palantir/seekio/InMemorySeekableDataInput.java [hunk 1]
```diff
@@ -31,12 +31,14 @@ public InMemorySeekableDataInput(byte[] bytes) {
 
     @Override
     public int read(byte[] buf, int offset, int length) throws IOException {
-        try {
+        int remaining = bytes.remaining();
+        if (remaining < length) {
+            bytes.get(buf, offset, remaining);
+            return remaining;
```
Comment (schlosna) [inline seek-io/src/main/java/com/palantir/seekio/InMemorySeekableDataInput.java]: If remaining is 0, shouldn't this return -1 (or throw EOF)? Probably need to clarify the spec and tests

Comment (markelliot) [reply seek-io/src/main/java/com/palantir/seekio/InMemorySeekableDataInput.java]: This is right because it returns number of bytes read.

**Review by markelliot — APPROVED** [review]

**Comment by ellisjoe** [comment]
@schlosna is right. `InputStream#read` javadoc:
```
        ... If no byte is available because the stream is at end of
     * file, the value <code>-1</code> is returned; otherwise, at least one
     * byte is read and stored into <code>b</code>.
```

File: seek-io/src/main/java/com/palantir/seekio/InMemorySeekableDataInput.java (hunk 1 above)
Comment (ellisjoe) [reply seek-io/src/main/java/com/palantir/seekio/InMemorySeekableDataInput.java]: done

**Pull request merged**